src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -4706,34 +4706,17 @@ public T[] Items
         {
             get
             {
-                List<T> tempList = new List<T>();
-                IEnumerator<T> currentEnumerator = _enumerable.GetEnumerator();
-
-                if (currentEnumerator != null)
-                {
-                    for (_count = 0; currentEnumerator.MoveNext(); _count++)
-                    {
-                        tempList.Add(currentEnumerator.Current);
-                    }
-                }
-                if (_count == 0)
+                T[] array = _enumerable.ToArray();
+                if (array.Length == 0)
                 {
                     throw new SystemCore_EnumerableDebugViewEmptyException();
                 }
-
-                _cachedCollection = tempList.ToArray();
-                return _cachedCollection;
+                return array;
             }
         }
 
         [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
         private IEnumerable<T> _enumerable;
-
-        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
-        private T[] _cachedCollection;
```

Comment:
Please try debugging in Visual Studio using this and make sure everything still works in the debug views, both on first access and subsequent access.  It's possible these fields are being used by the debugger, though hopefully not.